### PR TITLE
{math}[foss/2018b] Pyomo v5.5.0 w/ Python 2.7.15

### PR DESCRIPTION
**Pull request by smoors**
(created using `eb --new-pr`)

### CODE REVIEW

File: easybuild/easyconfigs/p/Pyomo/Pyomo-5.5.0-foss-2018b-Python-2.7.15.eb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+description = """ Pyomo is a Python-based open-source software package that supports a diverse set of optimization
+ capabilities for formulating and analyzing optimization models. """
+
+toolchain = {'name': 'foss', 'version': '2018b'}
+
+dependencies = [('Python', '2.7.15')]
+
+exts_defaultclass = 'PythonPackage'
+
+exts_download_dep_fail = True
```
Comment (boegel) [inline easybuild/easyconfigs/p/Pyomo/Pyomo-5.5.0-foss-2018b-Python-2.7.15.eb]: @smoors Please also set `use_pip = True`, you can do both at once with:

```python
exts_default_options = {
  'download_dep_fail': True,
  'use_pip': True,
}

Comment (smoors) [reply easybuild/easyconfigs/p/Pyomo/Pyomo-5.5.0-foss-2018b-Python-2.7.15.eb]: note to self: don't blindly copy code from @boegel without checking for style :)

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in this PR)
node2091.delcatty.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/6c5f0097a0713fddfc1225bb3225d977 for a full test report.

**Comment by boegel** [comment]
@smoors I guess it's worth reporting that issue upstream?

**Comment by boegel** [comment]
@smoors Hmm, it's quite strange there's a problem when installing `PyUtilib` with `pip`, since that's the officially recommended way of installing it, see https://github.com/PyUtilib/pyutilib.

Which `__init__` file is missing exactly?

**Comment by smoors** [comment]
@boegel its the base file: `pyutilib/__init__.py`
I suspect it occurs when installing with `--prefix`, similar to this:
https://github.com/hpcugent/vsc-install/issues/68

**Comment by boegel** [comment]
@smoors You're probably right, see also https://github.com/pypa/pip/issues/1924 where the `pip` maintainers basically say they won't fix this problem for Python packages using namespaces for Python 2.x & old Python 3.x versions.

Still, this should be reported upstream via https://github.com/PyUtilib/pyutilib/issues, to make sure they're aware of this problem, to ask what a proper workaround would be, and to hear their thoughts on it?

We can probably work around this easily by doing `touch %(installdir)s/lib/python2.7/site-packages/PyUtilib-*.egg/pyutilib/__init__.py` (but then we should also make sure `PyUtilib` is fully functional by doing additional sanity checks).

When installing with `setuptools`, the `pyutilib/__init__.py` that is installed is not empty, but the stuff in there (like `pkg_resources.declare_namespace`, etc.) is only relevant when you're spreading the `pyutilib` namespace across multiple different subdirectories (which is not the case when it's installed...).

**Comment by smoors** [comment]
@boegel I filed an issue, see https://github.com/PyUtilib/pyutilib/issues/39.

not only `pyutilib/__init__.py` is missing, also `pyutilib/component/__init__.py` and possibly others (I did not check).

since it works with setuptools, do you think it is really worth the effort to workaround and test just for pip?

**Comment by boegel** [comment]
@smoors We're pushing for using `pip` a bit because i) it's the recommended Python installation tool (see https://packaging.python.org/tutorials/installing-packages/), ii) we would like to start using it by default in `PythonPackage` soon (cfr. https://github.com/easybuilders/easybuild/issues/447).

This doesn't need to be a blocker for this PR though, since it's clear the problem is not on our side, but with `pip`.

File: easybuild/easyconfigs/p/Pyomo/Pyomo-5.5.0-foss-2018b-Python-2.7.15.eb [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        'checksums': ['9e5896d1372858f8dd3344faf4e5014d21849c756c8d5701f78f8a103b372d92'],
+    }),
+    ('ply', '3.11', {
+        'source_urls': ['https://pypi.python.org/packages/source/p/ply'],
+        'checksums': ['00c7c1aaa88358b9c765b6d3000c6eec0ba42abca5351b095321aef446081da3'],
+    }),
+    ('PyUtilib', '5.6.3', {
+        'source_urls': ['https://pypi.python.org/packages/source/p/pyutilib'],
+        'checksums': ['6a21fccfe691c39566c0bb19b5c9aa11bca8b076aa6f1dbf21c11711f5105191'],
+        'use_pip': False,
```
Comment (boegel) [inline easybuild/easyconfigs/p/Pyomo/Pyomo-5.5.0-foss-2018b-Python-2.7.15.eb]: @smoors Please add a comment above this line referencing to https://github.com/PyUtilib/pyutilib/issues/39 to clarify why this is required.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
gligar03.gligar.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/5b7ce4f769da52b8f73340faae930456 for a full test report.

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
node2723.swalot.os - Linux centos linux 7.5.1804, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz, Python 2.7.5
See https://gist.github.com/0e0ef6443b05637a2112f2561a561d80 for a full test report.

**Comment by boegel** [comment]
Going in, thanks @smoors!

**Pull request merged**